purchase_create_related/wizards/purchase_order_secondary_linker.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2025 Alberto Martínez <alberto.martinez@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, exceptions, fields, models
+
+
+class PurchaseOrderSecondaryLinker(models.TransientModel):
+    _name = "purchase.order.secondary.linker"
+    _description = "Wizard to link purchases to secondary purchases"
+
+    operation_type = fields.Selection(
+        selection=[
+            ("create", "Create Quotation Purchase"),
+            ("link", "Link to Quotation Purchase"),
+        ],
+        required=True,
+    )
+
+    main_purchase_ids = fields.Many2many(
+        string="Main Purchases",
+        comodel_name="purchase.order",
+        domain=[
+            ("main_purchase_ids", "=", False),
+        ],
+        required=True,
+    )
+    secondary_purchase_id = fields.Many2one(
+        string="Secondary Purchase",
+        comodel_name="purchase.order",
+        domain=[
+            ("secondary_purchase_ids", "=", False),
+            ("can_be_secondary", "!=", False),
+            ("state", "in", ["draft", "sent", "to approve"]),
+        ],
+    )
+    partner_id = fields.Many2one(
+        string="Secondary Purchase Partner",
+        comodel_name="res.partner",
+    )
+
+    def default_get(self, fields):
+        vals = super().default_get(fields)
+        main_purchase_ids = self.env["purchase.order"].browse(
+            self.env.context.get("main_purchase_ids")
```

Comment:
Be careful with the browse, because if the record does not exist it will instantiate it anyway. To ensure this, an exist() is used afterwards (you have examples in odoo)

Reply:
Done